Dockerfile

```diff
@@ -10,13 +10,13 @@ RUN dpkg --add-architecture i386
 # unless MLIR library is incompatible with current ROCm.
 
 RUN if [ "$USE_MLIR" = "ON" ] ; \
```

Comment:
Maybe we will get rid of this meaningless condition already

Reply:
@jerryyin I think MLIR wishes to retain the flexibility of using a different ROCm version? However indeed I would prefer to consolidate them.

Reply:
This branch condition was there because in the first MLIR release it is attached to a specific ROCm release.  (The enablement took our exclusive attention and supporting a newer ROCm become secondary priority.) Because of this, MIOpen is blocked by MLIR's old ROCm version. This condition make it such that MIOpen fall back to older ROCm release when testing MLIR related stages.

Agreed here that we can get rid of this condition because:
 - MLIR is always up to date with latest ROCm version, the risk of falling behind is minimal
 - MLIR has been tested in (almost) all MIOpen stage, this fallback will cause MIOpen to also effectively fallback to a previous ROCm (making the version update meaningless)